decode.go

```diff
@@ -112,21 +136,24 @@ func (dec *Decoder) SetBufferSize(n int) {
 	dec.buf = bytes.NewBuffer(make([]byte, 0, n))
 }
 
+// Get the file header
+func (dec *Decoder) Header() (*Header, error) {
+	// deserialize the file header
+	err := dec.readOSMHeader()
```

Comment:
Could this be changed to the following in order to keep the `err` variable scope as small as possible?
```
if err := dec.readOSMHeader(); err != nil {
    return nil, err
}
```